minibroker/minibroker-check.yaml

```diff
@@ -0,0 +1,451 @@
+  source:
+    <<: *s3-configs
+    versioned_file: ((s3-prefix))/checks/index.yaml
+
+- name: s3.minibroker-chart
+  type: s3
+  source:
+    <<: *s3-configs
+    regexp: ((s3-prefix))/checks/minibroker-(.*).tgz
+
+- name: s3.helm-charts-repo-index
+  type: s3
+  source:
+    <<: *helm-charts-s3-configs
+    versioned_file: ((helm-charts-s3-prefix))/checks/index.yaml
+
+- name: s3.redis-chart
+  type: s3
+  source:
+    <<: *helm-charts-s3-configs
+    regexp: ((helm-charts-s3-prefix))/checks/redis-(.*).tgz
+
+- name: s3.postgresql-chart
+  type: s3
+  source:
+    <<: *helm-charts-s3-configs
+    regexp: ((helm-charts-s3-prefix))/checks/postgresql-(.*).tgz
+
+- name: s3.mariadb-chart
+  type: s3
+  source:
+    <<: *helm-charts-s3-configs
+    regexp: ((helm-charts-s3-prefix))/checks/mariadb-(.*).tgz
+
+- name: s3.mongodb-chart
+  type: s3
+  source:
+    <<: *helm-charts-s3-configs
+    regexp: ((helm-charts-s3-prefix))/checks/mongodb-(.*).tgz
+
+groups:
+- name: minibroker
+  jobs:
+  - build-minibroker-image
+  - generate-minibroker-chart
+  - generate-minibroker-repo
+- name: charts
+  jobs:
+  - generate-mariadb
+  - generate-mongodb
+  - generate-postgres
+  - generate-redis
+  - generate-repo
+
+jobs:
+- name: build-minibroker-image
+  serial: true
+  plan:
+  - aggregate:
+    - get: src
+      trigger: true
+    - get: src-ci
+  - put: status
+    params:
+      context: build-minibroker
+      description: build-minibroker
+      path: src
+      state: pending
+  - task: build-minibroker
+    file: src-ci/minibroker/tasks/build-minibroker.yaml
+  - put: docker.minibroker
+    params:
+      build: minibroker/src/image
+    on_failure:
+      put: status
+      params:
+        context: build-minibroker
+        description: build-minibroker
+        path: src
+        state: failure
+    on_success:
+      put: status
+      params:
+        context: build-minibroker
+        description: build-minibroker
+        path: src
+        state: success
+
+
+- name: generate-minibroker-chart
+  plan:
+  - aggregate:
+    - get: src-ci
+    - get: src
+    - get: docker.minibroker
+      trigger: true
+      passed:
+      - build-minibroker-image
+  - put: status
+    params:
+      context: generate-minibroker-chart
+      description: generate-minibroker-chart
+      path: src
+      state: pending
+  - task: generate-chart
+    file: src-ci/minibroker/tasks/generate-minibroker-chart.yaml
+  - put: s3.minibroker-chart
+    params:
+      file: 'chart/minibroker-*.tgz'
+      acl: public-read
+    on_failure:
+      put: status
+      params:
+        context: generate-minibroker-chart
+        description: generate-minibroker-chart
+        path: src
+        state: failure
+    on_success:
+      put: status
+      params:
+        context: generate-minibroker-chart
+        description: generate-minibroker-chart
+        path: src
+        state: success
+- name: generate-minibroker-repo
+  plan:
+  - aggregate:
+    - get: src
```

Comment:
Is this used?

Reply:
It's used by the `status` to get the commit for the PR. The Github API is called to set the status for that commit.

Reply:
I think it's useful to have a comment there, it tripped me up as well when I was writing the pipeline.